{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "nvim version:\r\n\r\n```\r\nNVIM v0.2.0\r\nBuild type: Release\r\nCompilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -O2 -DNDEBUG -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.2.0/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include\r\n```\r\n\r\nVim (version: ) behaves differently? I don't use Vim.\r\n\r\nOperating system/version: Arch Linux\r\n\r\nTerminal name/version: xterm-256color\r\n\r\nWhen I am editing bash scripts and exits the script, the following error occurs:\r\n\r\n```\r\nError detected while processing function <SNR>107_PollServerReady[1]..<SNR>107_Pyeval[2]..provider#python3#Call:\r\nline   18:\r\nInvalid channel: 1\r\nError detected while processing function <SNR>107_PollFileParseResponse[1]..<SNR>107_Pyeval[2]..provider#python3#Call:\r\nline   18:\r\nInvalid channel: 1\r\n```\r\n\r\nList of plugins:\r\n\r\n```\r\nFinished. 0 error(s). \r\n[===================]\r\n\r\n- phpcomplete.vim: OK\r\n- vim_pygments_colorscheme: OK\r\n- YouCompleteMe: OK\r\n- vim-snippets: OK\r\n- Zenburn: OK\r\n- github-theme: OK\r\n- vim-aldmeris: OK\r\n- vim-phpfmt: OK\r\n- pw: OK\r\n- nerdtree: OK\r\n- vim-colors-codeschool: OK\r\n- vim-bufkill: OK\r\n- ultisnips: OK\r\n- supertab: OK\r\n- tcomment_vim: OK\r\n- vim-go: OK\r\n- syntastic: OK\r\n- vim-signature: OK\r\n- vim-airline: OK\r\n```\r\n",
    "closed_at": null,
    "closed_by": null,
    "comment_data": [
        {
            "author_association": "NONE",
            "body": "YouCompleteMe is the culprit...From what I was told it has to do with timers that was introduced in this PR. As far as I know its utterly harmless..\r\n\r\nhttps://github.com/Valloric/YouCompleteMe/issues/2636\r\n\r\nThe problem has nothing to do with neovim apparently, and I'm not aware of any workaround. I'm able to reproduce the problem by immediately exiting neovim as soon as I start it. Someone else was able to reproduce it by slowing down the server startup. I think its best to report the issue over there so they'll know.",
            "created_at": "2017-06-04T14:23:42Z",
            "html_url": "https://github.com/neovim/neovim/issues/6840#issuecomment-306043103",
            "id": 306043103,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/6840",
            "node_id": "MDEyOklzc3VlQ29tbWVudDMwNjA0MzEwMw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/306043103/reactions"
            },
            "updated_at": "2017-06-04T14:59:41Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/306043103",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/29156942?v=4",
                "events_url": "https://api.github.com/users/manipnip/events{/privacy}",
                "followers_url": "https://api.github.com/users/manipnip/followers",
                "following_url": "https://api.github.com/users/manipnip/following{/other_user}",
                "gists_url": "https://api.github.com/users/manipnip/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/manipnip",
                "id": 29156942,
                "login": "manipnip",
                "node_id": "MDQ6VXNlcjI5MTU2OTQy",
                "organizations_url": "https://api.github.com/users/manipnip/orgs",
                "received_events_url": "https://api.github.com/users/manipnip/received_events",
                "repos_url": "https://api.github.com/users/manipnip/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/manipnip/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/manipnip/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/manipnip",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "The python process is crashed.\r\nYou can get the crash log by below.\r\n\r\n```\r\n$ export NVIM_PYTHON_LOG_FILE=/tmp/log\r\n$ export NVIM_PYTHON_LOG_LEVEL=DEBUG\r\n$ nvim\r\ndo something\r\n```",
            "created_at": "2017-06-05T01:52:28Z",
            "html_url": "https://github.com/neovim/neovim/issues/6840#issuecomment-306085021",
            "id": 306085021,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/6840",
            "node_id": "MDEyOklzc3VlQ29tbWVudDMwNjA4NTAyMQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/306085021/reactions"
            },
            "updated_at": "2017-06-05T01:52:28Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/306085021",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/41495?v=4",
                "events_url": "https://api.github.com/users/Shougo/events{/privacy}",
                "followers_url": "https://api.github.com/users/Shougo/followers",
                "following_url": "https://api.github.com/users/Shougo/following{/other_user}",
                "gists_url": "https://api.github.com/users/Shougo/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/Shougo",
                "id": 41495,
                "login": "Shougo",
                "node_id": "MDQ6VXNlcjQxNDk1",
                "organizations_url": "https://api.github.com/users/Shougo/orgs",
                "received_events_url": "https://api.github.com/users/Shougo/received_events",
                "repos_url": "https://api.github.com/users/Shougo/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/Shougo/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/Shougo/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/Shougo",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "Stopping the timers on the `VimLeave` event seems to fix that issue. However, wouldn't it make sense for Neovim to automatically stop them before terminating the python process at exit?",
            "created_at": "2017-06-21T04:37:26Z",
            "html_url": "https://github.com/neovim/neovim/issues/6840#issuecomment-309960751",
            "id": 309960751,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/6840",
            "node_id": "MDEyOklzc3VlQ29tbWVudDMwOTk2MDc1MQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/309960751/reactions"
            },
            "updated_at": "2017-06-21T04:37:26Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/309960751",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/10026824?v=4",
                "events_url": "https://api.github.com/users/micbou/events{/privacy}",
                "followers_url": "https://api.github.com/users/micbou/followers",
                "following_url": "https://api.github.com/users/micbou/following{/other_user}",
                "gists_url": "https://api.github.com/users/micbou/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/micbou",
                "id": 10026824,
                "login": "micbou",
                "node_id": "MDQ6VXNlcjEwMDI2ODI0",
                "organizations_url": "https://api.github.com/users/micbou/orgs",
                "received_events_url": "https://api.github.com/users/micbou/received_events",
                "repos_url": "https://api.github.com/users/micbou/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/micbou/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/micbou/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/micbou",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "NONE",
            "body": "same issue here, got the python crash log like you said:\r\n\r\n```\r\n2018-03-13 16:29:33,055 [DEBUG @ session.py:response_cb:153] 58788 - response is available for greenlet <greenlet.greenlet object at 0x10adff3d8>, switching back\r\n2018-03-13 16:29:33,055 [WARNING @ async_session.py:_on_message:83] 58788 - Traceback (most recent call last):\r\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/async_session.py\", line 80, in _on_message\r\n    self._handlers.get(msg[0], self._on_invalid_message)(msg)\r\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/async_session.py\", line 101, in _on_response\r\n    self._pending_requests.pop(msg[1])(msg[2], msg[3])\r\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/session.py\", line 154, in response_cb\r\n    gr.switch(err, rv)\r\ngreenlet.error: cannot switch to a different thread\r\n\r\n2018-03-13 16:29:33,055 [DEBUG @ msgpack_stream.py:send:34] 58788 - sent [1, 0, 'Traceback (most recent call last):\\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/async_session.py\", line 80, in _on_message\\n    self._handlers.get(msg[0], self._on_invalid_message)(msg)\\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/async_session.py\", line 101, in _on_response\\n    self._pending_requests.pop(msg[1])(msg[2], msg[3])\\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/session.py\", line 154, in response_cb\\n    gr.switch(err, rv)\\ngreenlet.error: cannot switch to a different thread\\n', None]\r\n2018-03-13 16:29:33,055 [DEBUG @ base.py:send:117] 58788 - Sending 'b'\\x94\\x01\\x00\\xda\\x02uTraceback (most recent call last):\\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/async_session.py\", line 80, in _on_message\\n    self._handlers.get(msg[0], self._on_invalid_message)(msg)\\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/async_session.py\", line 101, in _on_response\\n    self._pending_requests.pop(msg[1])(msg[2], msg[3])\\n  File \"/Users/timfeirg/.virtualenvs/citadel/lib/python3.6/site-packages/neovim/msgpack_rpc/session.py\", line 154, in response_cb\\n    gr.switch(err, rv)\\ngreenlet.error: cannot switch to a different thread\\n\\xc0''\r\n2018-03-13 16:29:33,055 [DEBUG @ msgpack_stream.py:_on_data:59] 58788 - waiting for message...\r\n2018-03-13 16:29:33,055 [DEBUG @ msgpack_stream.py:_on_data:64] 58788 - unpacker needs more data...\r\n2018-03-13 16:29:33,055 [DEBUG @ base.py:_on_error:175] 58788 - 32\r\n2018-03-13 16:29:33,055 [DEBUG @ base.py:stop:153] 58788 - Stopped event loop\r\n2018-03-13 16:29:33,056 [DEBUG @ base.py:run:144] 58788 - Exited event loop\r\n```\r\n\r\n@Shougo ",
            "created_at": "2018-03-14T03:23:00Z",
            "html_url": "https://github.com/neovim/neovim/issues/6840#issuecomment-372892089",
            "id": 372892089,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/6840",
            "node_id": "MDEyOklzc3VlQ29tbWVudDM3Mjg5MjA4OQ==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/372892089/reactions"
            },
            "updated_at": "2018-03-14T03:23:00Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/372892089",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/4319104?v=4",
                "events_url": "https://api.github.com/users/timfeirg/events{/privacy}",
                "followers_url": "https://api.github.com/users/timfeirg/followers",
                "following_url": "https://api.github.com/users/timfeirg/following{/other_user}",
                "gists_url": "https://api.github.com/users/timfeirg/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/timfeirg",
                "id": 4319104,
                "login": "timfeirg",
                "node_id": "MDQ6VXNlcjQzMTkxMDQ=",
                "organizations_url": "https://api.github.com/users/timfeirg/orgs",
                "received_events_url": "https://api.github.com/users/timfeirg/received_events",
                "repos_url": "https://api.github.com/users/timfeirg/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/timfeirg/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/timfeirg/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/timfeirg",
                "user_view_type": "public"
            }
        },
        {
            "author_association": "CONTRIBUTOR",
            "body": "Hm.  It seems greenlet error.",
            "created_at": "2018-03-14T03:30:15Z",
            "html_url": "https://github.com/neovim/neovim/issues/6840#issuecomment-372893283",
            "id": 372893283,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/6840",
            "node_id": "MDEyOklzc3VlQ29tbWVudDM3Mjg5MzI4Mw==",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/372893283/reactions"
            },
            "updated_at": "2018-03-14T03:30:15Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/372893283",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/41495?v=4",
                "events_url": "https://api.github.com/users/Shougo/events{/privacy}",
                "followers_url": "https://api.github.com/users/Shougo/followers",
                "following_url": "https://api.github.com/users/Shougo/following{/other_user}",
                "gists_url": "https://api.github.com/users/Shougo/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/Shougo",
                "id": 41495,
                "login": "Shougo",
                "node_id": "MDQ6VXNlcjQxNDk1",
                "organizations_url": "https://api.github.com/users/Shougo/orgs",
                "received_events_url": "https://api.github.com/users/Shougo/received_events",
                "repos_url": "https://api.github.com/users/Shougo/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/Shougo/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/Shougo/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/Shougo",
                "user_view_type": "public"
            }
        }
    ],
    "comments": 5,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/6840/comments",
    "created_at": "2017-06-03T22:52:36Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/6840/events",
    "html_url": "https://github.com/neovim/neovim/issues/6840",
    "id": 233399933,
    "labels": [
        {
            "color": "c5def5",
            "default": false,
            "description": null,
            "id": 144672485,
            "name": "provider",
            "node_id": "MDU6TGFiZWwxNDQ2NzI0ODU=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/provider"
        },
        {
            "color": "c7def8",
            "default": false,
            "description": "channels, RPC, msgpack",
            "id": 242522707,
            "name": "channels-rpc",
            "node_id": "MDU6TGFiZWwyNDI1MjI3MDc=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/channels-rpc"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/6840/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": null,
        "closed_issues": 686,
        "created_at": "2014-05-10T20:43:04Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
            "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
            "followers_url": "https://api.github.com/users/justinmk/followers",
            "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
            "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/justinmk",
            "id": 1359421,
            "login": "justinmk",
            "node_id": "MDQ6VXNlcjEzNTk0MjE=",
            "organizations_url": "https://api.github.com/users/justinmk/orgs",
            "received_events_url": "https://api.github.com/users/justinmk/received_events",
            "repos_url": "https://api.github.com/users/justinmk/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/justinmk",
            "user_view_type": "public"
        },
        "description": "Low priority items. We plan to work on this but don't have a target date.",
        "due_on": null,
        "html_url": "https://github.com/neovim/neovim/milestone/6",
        "id": 655037,
        "labels_url": "https://api.github.com/repos/neovim/neovim/milestones/6/labels",
        "node_id": "MDk6TWlsZXN0b25lNjU1MDM3",
        "number": 6,
        "open_issues": 635,
        "state": "open",
        "title": "backlog",
        "updated_at": "2025-01-21T00:09:37Z",
        "url": "https://api.github.com/repos/neovim/neovim/milestones/6"
    },
    "node_id": "MDU6SXNzdWUyMzMzOTk5MzM=",
    "number": 6840,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/neovim/neovim/issues/6840/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "sub_issues_summary": {
        "completed": 0,
        "percent_completed": 0,
        "total": 0
    },
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/6840/timeline",
    "title": "Error detected while processing function <SNR>107_PollServerReady[",
    "updated_at": "2025-01-19T10:46:32Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/6840",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/1302896?v=4",
        "events_url": "https://api.github.com/users/greencopper/events{/privacy}",
        "followers_url": "https://api.github.com/users/greencopper/followers",
        "following_url": "https://api.github.com/users/greencopper/following{/other_user}",
        "gists_url": "https://api.github.com/users/greencopper/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/greencopper",
        "id": 1302896,
        "login": "greencopper",
        "node_id": "MDQ6VXNlcjEzMDI4OTY=",
        "organizations_url": "https://api.github.com/users/greencopper/orgs",
        "received_events_url": "https://api.github.com/users/greencopper/received_events",
        "repos_url": "https://api.github.com/users/greencopper/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/greencopper/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/greencopper/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/greencopper",
        "user_view_type": "public"
    }
}